feature/preference/src/main/java/io/github/droidkaigi/confsched2020/preference/ui/PreferencesFragment.kt

```diff
@@ -18,30 +19,34 @@ import dagger.Provides
 import io.github.droidkaigi.confsched2020.App
 import io.github.droidkaigi.confsched2020.di.AppComponent
 import io.github.droidkaigi.confsched2020.di.PageScope
+import io.github.droidkaigi.confsched2020.ext.assistedViewModels
 import io.github.droidkaigi.confsched2020.preference.R
+import io.github.droidkaigi.confsched2020.preference.ui.di.PreferenceAssistedInjectModule
+import io.github.droidkaigi.confsched2020.preference.ui.viewmodel.PreferenceViewModel
+import javax.inject.Inject
+import javax.inject.Provider
 
 class PreferencesFragment : PreferenceFragmentCompat() {
 
-    private val darkThemeSwitchChangeListener =
-        Preference.OnPreferenceChangeListener { _, newValue ->
-            AppCompatDelegate.setDefaultNightMode(
-                if (newValue as Boolean) {
-                    MODE_NIGHT_YES
-                } else {
-                    MODE_NIGHT_NO
-                }
-            )
-            (activity as? AppCompatActivity)?.delegate?.applyDayNight()
-            return@OnPreferenceChangeListener true
-        }
+    @Inject
+    lateinit var preferenceModelFactory: Provider<PreferenceViewModel>
+    private val preferenceViewModel by assistedViewModels {
+        preferenceModelFactory.get()
+    }
+
+    private var isNightMode = false
```

Comment:
Thanks!
Can we remove this property by bringing findPreference to onViewCreated👀

Reply:
Thanks for the feedback and that was my first thought as well.
I thought the value of switchPreference should be set before creating the view thus I keep it in the onCreatePreference method, but apparently I was wrong 😅

Reply:
I didn't try it. Can you try it ?🙏

Reply:
I was trying the change and testing on different android version
I realized that as letting observer to handle the change of dark theme would cause weird behavior when user switch on/off dark theme having the app running in Android 10

In fact Android 10 handles the switch on/off event quite nicely

Therefore I’m considering for specifically this setting, `nightMode` could be tackle with viewmodel for SDK Version below 28 and for versions after 29, we could leave it as it is

The viewmodel could still be helpful when adding new options later on though

Reply:
In other words, is it OK to work with the following changes?
https://github.com/DroidKaigi/conference-app-2020/pull/365/commits/056c2ff6247fbe36c8e2883b88914c5d088e7bfa

Reply:
I will push the latest commit soon

Reply:
Thanks!

Reply:
I’ve pushed the commit yesterday
Please give some advice when you are free as there should be a lot of other PR going right now😉